core/runtime.rs

```diff
@@ -510,6 +513,46 @@ impl JsRuntime {
         let context =
           bindings::initialize_context(scope, &op_ctxs, snapshot_options);
 
+        {
+          let mut scope = v8::ContextScope::new(scope, context);
+          match scope.get_context_data_from_snapshot_once::<v8::Object>(0) {
+            Ok(val) => {
+              let next_module_id = {
+                let next_module_id_str =
+                  v8::String::new(&mut scope, "next_module_id").unwrap();
+                let next_module_id =
+                  val.get(&mut scope, next_module_id_str.into()).unwrap();
+                assert!(next_module_id.is_int32());
+                let integer = next_module_id.to_integer(&mut scope).unwrap();
+                integer.int32_value(&mut scope).unwrap()
+              };
+              let no_of_modules = next_module_id - 1;
+
+              for i in 1..=no_of_modules {
+                let result = scope
+                  .get_context_data_from_snapshot_once::<v8::Module>(
+                    i as usize,
+                  );
+                // FIXME(bartlomieju): gracefully handle error
+                let module_local = result.unwrap();
+                let module_global = v8::Global::new(&mut scope, module_local);
+                module_handles.push(module_global);
+              }
+
+              let global = v8::Global::new(&mut scope, val);
+              module_map_data = Some(global);
+            }
+            Err(err) => match err {
+              DataError::BadType { actual, expected } => {
+                dbg!(actual, expected);
+              }
+              DataError::NoData { expected } => {
+                dbg!(expected);
+              }
+            },
```

Comment:
This should be a helper function that returns `T` and panics if we hit `DataError`